libs/designsystem/src/lib/components/modal/services/modal-navigation.service.ts

```diff
@@ -0,0 +1,197 @@
+  }
+
+  private getModalRoutePath(route: Route, parentPath: string[]): string[] {
+    const modalOutletName = 'modal';
+    if (!!route.path && route.outlet === modalOutletName) {
+      const modalOutletPath = `(${modalOutletName}:${route.path})`;
+      const modalRoutePath = [...parentPath, modalOutletPath].join('/');
+      return [modalRoutePath];
+    }
+    const currentPath = [...parentPath];
+    if (!!route.path) {
+      currentPath.push(route.path);
+    }
+    return ([] as string[]).concat(...this.getModalRoutePaths(route.children, currentPath));
+  }
+
+  private getModalRoutePaths(routes: Routes, parentPath: string[]): string[] {
+    return Array.isArray(routes)
+      ? ([] as string[]).concat(...routes.map((route) => this.getModalRoutePath(route, parentPath)))
+      : [];
+  }
+
+  private isNewModalWindow(navigationEnd: NavigationEnd) {
+    const currentNavigation = this.router.getCurrentNavigation();
+    if (!currentNavigation || !currentNavigation.previousNavigation) {
+      return true;
+    }
+    const previousRoutePath = (
+      currentNavigation.previousNavigation.finalUrl ||
+      currentNavigation.previousNavigation.extractedUrl
+    ).toString();
+    const wasModalRoute = this.isModalRoute(previousRoutePath);
+    const isModalRoute = this.isModalRoute(navigationEnd.urlAfterRedirects);
+    if (!wasModalRoute && isModalRoute) {
+      return true;
+    }
+
+    const currentModalRouteParent = navigationEnd.urlAfterRedirects.split('/(modal:')[0];
+    const previousModalRouteParent = previousRoutePath.split('/(modal:')[0];
+
+    return previousModalRouteParent !== currentModalRouteParent;
+  }
+
+  private navigationEndListener$ = this.router.events.pipe(
+    filter((event): event is NavigationEnd => event instanceof NavigationEnd)
+  );
+
+  modalRouteActivated$ = this.navigationEndListener$.pipe(
+    map((event) => this.isModalRoute(event.urlAfterRedirects)),
+    filter((isModalRoute) => isModalRoute),
+    map(() => this.getCurrentActivatedRoute())
+  );
+
+  modalRouteDeactivatedFor(routeConfig: Routes[]) {
+    if (Array.isArray(routeConfig)) {
+      const modalRouteMap = this.getModalRouteMap(routeConfig);
+      const hasModalRoutes = modalRouteMap.size > 0;
+      if (hasModalRoutes) {
+        return this.navigationEndListener$.pipe(
+          pairwise(),
+          filter(([prevNavigation, _]) => this.isModalRoute(prevNavigation.urlAfterRedirects)), // Only emit if previous route was modal
+          map(([_, currentNavigation]) => {
+            const isNewModalRoute = this.isModalRoute(currentNavigation.urlAfterRedirects);
+            // Deactivate modal route if new route is NOT modal OR is outside current parent route:
+            return !isNewModalRoute || this.isNewModalWindow(currentNavigation);
+          }),
+          filter((isDeactivation) => isDeactivation)
+        );
+      }
+    }
+    return EMPTY;
+  }
+
+  modalRouteActivatedFor(routeConfig: Routes[]) {
```

Comment:
not sure "modalRouteActivatedFor" is meaningful to me, at least it requires some context.

Reply:
The `for` hints to the param `routerConfig` - so it reads "when a modal route is activated for this given config" - or at least, thats the idea ;)